content/setting-up-services/how-to-set-up-services/leverage-service-commands.md

```diff
@@ -53,3 +68,55 @@ services:
 
 Notice that each stage is optional for a given service. There may not be any commands required for that service during
 some stage. When that is the case, the stage can be excluded completely.
+
+## Service commands to run after Preview build
+
+You can also configure service commands to run _after_ a Tugboat Preview has built. The three service commands that run
+after the build process are:
+
+- **start** - Run commands every time a Preview container starts. This happens any time the container starts; after a
+  Preview has [suspended](/building-a-preview/preview-deep-dive/how-previews-work#status-message) or been
+  [stopped](/building-a-preview/administer-previews/change-preview-states#start-stop); or if you
+  [Reset](/building-a-preview/administer-previews/change-preview-states#reset) the Preview. `start` might be useful for
+  [running a background process](../running-a-background-process/); to warm up caches, for example.
+- **online** - Run commands once, after a Preview build has completed. These commands only run after the Preview
+  snapshot, when a Preview has been through **Build**, **Rebuild** or **Refresh**. For more info on the build process
+  and build snapshots, see: [How Previews Work](/building-a-preview/preview-deep-dive/how-previews-work/).
+- **clone** - Run commands on the cloned (new) Preview that has been created from another Preview. Commands that you run
+  in `clone` get committed as part of the build snapshot, so things like Resetting an image do not erase `clone`
+  commands. For more info, see: [How Previews Work](/building-a-preview/preview-deep-dive/how-previews-work/).
+
+For an example of the `online` command, here's a sample code snippet from
+[our Diffy integration](/starter-configs/code-snippets/diffy-integration/):
+
+```yaml
```

Comment:
Thoughts on truncating this to only include the `online:` section? It would make the example more focused, and give us less example code to maintain

Reply:
I wanted to include more than one service command, so users could see `online` in the context of a larger config - i.e. alongside other service commands (`init`). I'm fine with stripping out some of the comments or something if you feel there's value in that. But I do want to keep some context beyond just the "online" section.

Reply:
How about we leave out `init` here, since it is covered above, and include the memcache `start` example from below. That leaves each section with a list of commands, what they do, and an example for each

Reply:
For example:

```
services:
   apache:
     image: tugboatqa/httpd:2.4
     default: true
     commands:
      start:
        # Warm the cache
         - sudo -u www-data /var/lib/tugboat/vendor/bin/drush --root /var/lib/tugboat/web warmer:enqueue -l localhost
           --verbose --run-queue
       online:
         # Compare this service with production using Diffy
         - diffy project:compare $DIFFY_PROJECT_ID prod custom --env2Url=$TUGBOAT_SERVICE_URL
```

Reply:
I specifically wanted to show commands that span the build and after-build command context, to show how those might work together in the same config. I thought that information would be helpful for users who are new to Tugboat, or who are very familiar with commands in the build context but be wondering whether implementing post-build commands works the same way. 

I'm fine with combining `start` and `online` if you think it makes sense vs. maintaining these separately, but would like to include _some_ build-phase command context. Thoughts?

Reply:
I'd like to include a `start` example smewhere, otherwise it's the only one left out.

I don't have strong opinions about the rest, was mostly just brainstorming how to cleanly add a `start` example.

Reply:
Ok. I'll add `start` per this example, and I'll leave `init` but pare it down so there's less clutter and it's more just "here are some commands in `init`.